pkg/api/types.go

```diff
@@ -205,6 +205,8 @@ type VolumeSource struct {
 	Glusterfs *GlusterfsVolumeSource `json:"glusterfs"`
 	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace
 	PersistentVolumeClaimVolumeSource *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
+	// Metadata represents an host volume which contains pod's metadata
```

Comment:
I'm not loving the 'host volume' wording here.  How about `Metadata represents metadata about the pod that should populate this volume`?